### DISCUSSION TITLE
how to get entire tailwind library minified with cli?

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#12561] [Help]
Hi, I am using tailwind cli to produce minified file, but however I would like it to include entire tailwind classes in minified version whether present in my project or not? how can I achieve it or where can I get latest version fully minified tailwind? Thanks

**Comment by wongjn** [comment]
This would not really be possible for v3 – there's an infinite amount of classes possible (due to the arbitrary features) and even without these, if you really wanted every single class possible, you would need to generate every possible combination of the variants, (like `dark:first:hover:focus:md:visited:last:only:…:m-1`) so there would be all the "base" classes plus all these permutations of variants and thus the total number of possible classes would be something like (70![^1] + 1) × number of "base" classes (maths probably not correct here, but something like this). The "real" number would be slightly less since some variants don't make sense together, i.e. `sm:md:lg:xl:m-1` but this obscenely large number should demonstrate the point.

However, if you do want to do this anyway, you'd use a [regular expression `safelist`](https://tailwindcss.com/docs/content-configuration#using-regular-expressions) like:

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  safelist: [
    {
      pattern: /^/,
      variants: [
        'sm',
        'md',
        'lg',
        'hover:',
        'sm:hover:',
        // …
      ],
    },
  ],
  // …
}
```

[^1]: 70 factorial, equal to 11978571669969891796072783721689098736458938142546425857555362864628009582789845319680000000000000000

**Reply by ghost** [reply]
wow, didn't take this into consideration for sure. Thank you for letting me know about this :)
